Source/WebCore/page/FocusController.cpp

```diff
@@ -572,13 +622,23 @@ Element* FocusController::findFocusableElementAcrossFocusScope(FocusDirection di
 {
     ASSERT(!is<Element>(currentNode) || !isNonFocusableScopeOwner(downcast<Element>(*currentNode), event));
 
-    if (currentNode && direction == FocusDirection::Forward && is<Element>(currentNode) && isFocusableScopeOwner(downcast<Element>(*currentNode), event)) {
-        if (Element* candidateInInnerScope = findFocusableElementWithinScope(direction, FocusNavigationScope::scopeOwnedByScopeOwner(downcast<Element>(*currentNode)), 0, event))
-            return candidateInInnerScope;
+    if (currentNode && direction == FocusDirection::Forward && is<Element>(currentNode)) {
+        if (isFocusableScopeOwner(downcast<Element>(*currentNode), event)) {
+            if (Element* candidateInInnerScope = findFocusableElementWithinScope(direction, FocusNavigationScope::scopeOwnedByScopeOwner(downcast<Element>(*currentNode)), 0, event))
+                return candidateInInnerScope;
+        } else if (isOpenPopoverInvoker(currentNode)) {
+            if (Element* candidateInInnerScope = findFocusableElementWithinScope(direction, FocusNavigationScope::scopeOwnedByPopoverInvoker(downcast<const HTMLFormControlElement>(*currentNode)), nullptr, event))
```

Comment:
The fact currentNode is a. HTMLFormControlElement isn't obvious from the check.
I suggest we make isOpenPopoverInvoker return a casted HTMLFromControlElement instead.

Reply:
Done.